Assets/Scripts/Entity/Entity.cs

```diff
@@ -19,6 +20,13 @@ public abstract class Entity : MonoBehaviour
     // Generic reference to whatever weapon the entity uses for firing
     protected IWeapon fireableWeapon;
 
+    public void setisDead(){
+        gameObject.tag = "Dead";
+        gameObject.GetComponent<Collider>().isTrigger = true;
+        gameObject.GetComponent<CapsuleCollider>().radius = .7f;
```

Comment:
This is known as a "magic number" and is an anti-pattern. In this case you are hard coding the ENTITY's trigger collider width on death. Also, this implementation is applying to ANY entity, whether it be player, enemy, or anything else we add as an entity. Since this is removing the game object's tag, this may cause extend-ability issues (tie our hands) later if we decide to add some sort of post death proximity detection for example. Not to mention this is also tying our hands and forcing every entity that is killable to have a collider, which may not be an issue but just want to point out.

You don't have to redesign this if you dont want to or cant figure out a better design. Just, at the very least, what to make sure you start to see how this (and some of our other code) can be restricting in terms of extend-ability.

Reply:
Actually, I might not need to set the radius to .7 here. I think the best radius is .7 (played around with the radius for a bit) for both the player and the enemy (you can get real close otherwise and the player may think they are colliding but they are not). I suggest you try out the radius for the cylinder collider (for both enemy and player) and confirm whether you think .7 is best.

Reply:
What would be an example of post death proximity detection?

Reply:
Well Alek wants the player to pass through dead enemies. How else should we provide for both a mesh and a "pass through ability"?

Reply:
Nah, I kinda liked how we had it in the demo where we couldn't pass through enemies. Up to you guys.

Reply:
Oh shit, you right boss. I didnt realize the demo had it that way.